packages/@aws-cdk/aws-lambda-event-sources/lib/dynamodb.ts

```diff
@@ -9,6 +9,8 @@ export interface DynamoEventSourceProps extends StreamEventSourceProps {
  * Use an Amazon DynamoDB stream as an event source for AWS Lambda.
  */
 export class DynamoEventSource extends StreamEventSource {
+  private _eventSourceMappingId: string | undefined = undefined;
```

Comment:
Use `?` to declare optional types. Same in other places.

```suggestion
  private _eventSourceMappingId?: string;
```